xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js

```diff
@@ -71,7 +71,15 @@ define([
         urlParams[pair[0]] = pair[1];
       }
     }
-    return $.post(new URL('?', url), $.param($.extend(urlParams, data), true));
+    return $.post(new URL('?', url), $.param($.extend(urlParams, data), true))
+      .then((nodes) => {
+        nodes.forEach((object) => {
+          // Each node will be rendered as an HTML element.
+          // As such, we want to make sure we escape space characters in their ids
+          object.id = object.id.replaceAll(' ','%20');
```

Comment:
```suggestion
          object.id = object.id.replaceAll(' ', '%20');
```

Are we sure that there is no `%20` in the id attribute? If not, shouldn't we also replace `%` by `%25`? Alternatively, would it be possible to store the original id, e.g., in a data attribute and use that as the id parameter in the request?

Reply:
Somehow testing manually this pitfall doesn't bring up any error... I don't understand why so there's probably something that needs to be fixed somewhere anyway :thinking:

Reply:
Okay, the test I conducted yesterday did not fail because I misremembered what I should test. After taking another look, the getChildren can get broken for page titles with a `%20` :+1:

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/3015/commits/31648cabdc7ad4ec206cb7ae48b954c8d78dad2c :+1: 

![18356-afterChanges](https://github.com/xwiki/xwiki-platform/assets/28761965/ad549ef9-0757-449f-9f20-ea5b77280025)
We can see on the screenshot above (taken with the latest version of this PR) that page titles with `%20` and `%25` as well as other page titles all fulfill the following:
* The IDs of the `a` and `li` elements displaying them are properly escaped: there's no space inside.
* The request to get their children is successful: the tree is generated properly and entirely.